gobblin-runtime/src/main/java/org/apache/gobblin/runtime/SafeDatasetCommit.java

```diff
@@ -181,6 +190,14 @@ public Void call()
     return null;
   }
 
+  private void submitFailureEvent(JobState.DatasetState datasetState) {
```

Comment:
rename to `maybeSubmitFailureEvent`. As it is, it gives the impression it always sends the event.

Reply:
+1